build/BuildSteps.cs

```diff
@@ -471,6 +477,35 @@ public static List<string> GetUnsignedBinaries(string targetDir)
             return unSignedPackages;
         }
 
+        private static List<string> GetDelaySignedBinaries(string targetDir)
+        {
+            // This will only work in Windows and assumes that "sn" tool is present in this expected location.
+            // We are ok doing this right now as this doesn't run in a normal build scenario and only when we need to validate
+            // signing. This only happens today in our release pipeline.
+            string snLocation = "C:\\Program Files (x86)\\Microsoft SDKs\\Windows\\v10.0A\\bin\\Netfx 4.8 Tools\\sn.exe";
+
+            Console.WriteLine($"Checking if any assemblies in '{targetDir} are delay-signed or test-signed using tool '{snLocation}'");
+
+            string[] files = Directory.GetFiles(targetDir, "*", SearchOption.AllDirectories);
+            var delaySigned = new List<string>();
+
+            foreach (string file in files)
+            {
+                var commandOutput = Shell.GetOutput(snLocation, $" -q -vf {file}", ignoreExitCode: true);
+                if (commandOutput.Contains("delay-signed", StringComparison.OrdinalIgnoreCase)
```

Comment:
nit: move `delay-signed` and `test-signed` to constants

Reply:
Thanks! To keep it consistent, I moved it as a static private string at the top of the file, as that's where we had the one other constant.